data-protocols/ids/ids-transform-v1/src/main/java/org/eclipse/dataspaceconnector/ids/transform/type/asset/AssetFromIdsRepresentationTransformer.java

```diff
@@ -51,29 +52,32 @@ public Class<Asset> getOutputType() {
             return null;
         }
 
-        IdsId idsId;
-        try {
-            idsId = IdsIdParser.parse(object.getId().toString());
-        } catch (IllegalArgumentException e) {
-            context.reportProblem(String.format("cannot read IdsId from representation (id: %s)", object.getId().toString()));
-            return null;
+        var result = IdsId.from(object.getId());
+        if (result.failed() || result.getContent() == null) {
```

Comment:
Actually, if the result is failed, the content is always null and if the result succeeded the content is always non null (except with the `Result<Void>` type). Same for all the other checks.

```suggestion
        if (result.failed()) {
```